common/lib/dependabot/pull_request_creator/message_builder/link_and_mention_sanitizer.rb

```diff
@@ -14,6 +14,7 @@ class LinkAndMentionSanitizer
           github\.com/(?<repo>#{GITHUB_USERNAME}/[^/\s]+)/
           (?:issue|pull)s?/(?<number>\d+)
         }x.freeze
+        GITHUB_NWO_REGEX = %r{(?<repo>#{GITHUB_USERNAME}/[^/\s#]+)#(?<number>\d+)}.freeze
```

Comment:
Could you explain this regex?

Reply:
I cribbed it together from the above regex that grabs the repo and number from a full github pr or issue link.  Given a string like `myorg/myrepo#123` the regex will match myorg based on the GITHUB_USERNAME regex, then will match nonwhitespace and non # chars after a / as the repo name, then grab the digits after the # as the issue or PR number.

Reply:
@pavera : could you put the regex explanation in the comment as well. It will be easier for future debugging.

Reply:
I'd recommend a positive lookahead instead of matching ("not character class" then "character class") 

```suggestion
        GITHUB_NWO_REGEX = %r{(?<repo>#{GITHUB_USERNAME}/.+)(?=#(?<number>\d+))}.freeze
```